src/libs/entropyGenerator/entropyGenerator.ts

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable no-bitwise */
+import { getBytes, keccak256, randomBytes } from 'ethers'
+
+export class EntropyGenerator {
+  #entropyPool: Uint8Array = new Uint8Array(0)
+
+  generateRandomBytes(length: number, extraEntropy: string): Uint8Array {
+    this.#resetEntropyPool()
+    this.#collectSystemNoiseEntropy()
+    this.#collectTimeEntropy()
+
+    if (extraEntropy) {
+      const encoder = new TextEncoder()
+      const uint8Array = encoder.encode(extraEntropy)
+      this.addEntropy(uint8Array)
+    }
+
+    if (this.#entropyPool.length === 0) throw new Error('Entropy pool is empty')
+
+    const hash = getBytes(keccak256(this.#entropyPool))
+    const randomBytesGenerated = randomBytes(length)
+    // ensures non-deterministic final output
```

Comment:
this comment is incorrect - not sure what you meant but it is deterministic - the same input will always generate the same output